src/global/signal_handler.cc

```diff
@@ -152,6 +152,30 @@ void install_standard_sighandlers(void)
 #include "common/Thread.h"
 #include <errno.h>
 
+void get_name_by_pid(pid_t pid, char *task_name)
+{
+  char proc_pid_path[PATH_MAX] = {0};
+  snprintf(proc_pid_path, PATH_MAX, "/proc/%d/cmdline", pid);
+  int file_fd = open(proc_pid_path, O_RDONLY);
+
+  if (file_fd < 0) {
+    derr << "Failure to open '" << proc_pid_path 
+         << "' error = " << file_fd 
```

Comment:
usually we do
int r = errno;
...

and then use cpp_strerror(r) to get the full error string. you may have to include common/errno.h.